stdlib/entity/resource.lua

```diff
@@ -0,0 +1,81 @@
+--- Resource utilities.
+-- @module Resource
+-- @usage local Resource = require('stdlib/entity/resource')
+
+local Surface = require 'stdlib/area/surface'
+local Tile = require 'stdlib/area/tile'
+local Queue = require 'stdlib/queue'
+
+Resource = {} --luacheck: allow defined top
+
+--- Get the resource entity at the specified position and surface.
+-- Adapted from YARM/resmon.lua -> find_resource_at
+-- @tparam ?|string|LuaSurface surface the surface of the position
+-- @tparam LuaPosition position the position to check
+-- @treturn LuaEntity|nil the resource entity or nil if there is no resource there
+function Resource.get_resource_at(surface, position)
+    local surfaces = Surface.lookup(surface)
+
+    if #surfaces == 0 then
+        return nil
+    end
+
+    local tile_at_position = Tile.from_position(position)
+    local tile_area = Tile.to_area(tile_at_position)
+
+    local resource_at_tile = surface.find_entities_filtered{area=tile_area, type='resource'}
+
+    if #resource_at_tile < 1 then
+        return nil
+    end
+
+    return resource_at_tile[1] -- there should never be another resource at the exact same coordinates
+end
+
+--- From the resource at the given surface and position, return all connected (horizontally, vertically and diagonally) resource entities of the same type.
+-- (For now, just ore patches. Problems arise when a single resource entity spans multiple tiles).
+-- @tparam LuaSurface surface the surface of the position
+-- @tparam LuaPosition position the position to check
+-- @treturn {LuaEntity*} a table containing the resource entity at the given position and all connected resource entities of the same type or an empty table if there is no resource there
```

Comment:
Everyone likes regex :) except lua and ldoc :)
https://stevedonovan.github.io/ldoc/manual/doc.md.html#Tag_Modifiers